godot_project/autoloads/data/color_palettes/periodic_table_color_palette.gd

```diff
@@ -0,0 +1,183 @@
+@tool
+class_name PeriodicTableColorPalette extends Resource
+
+@export var _data: Dictionary = _create_default_data()
+var _page: int = 1
+
+func _init() -> void:
+	_page = 1
+
+
+func get_color_for_element(in_atomic_number: int) -> Color:
+	return _get_property_for_element(in_atomic_number, "color")
+
+
+func get_noise_color_for_element(in_atomic_number: int) -> Color:
+	return _get_property_for_element(in_atomic_number, "noise_color")
+
+
+func get_bond_color_for_element(in_atomic_number: int) -> Color:
+	return _get_property_for_element(in_atomic_number, "bond_color")
+
+
+func get_font_color_for_element(in_atomic_number: int) -> Color:
+	return _get_property_for_element(in_atomic_number, "font_color")
+
+
+func _get_property_for_element(in_atomic_number: int, in_property: String) -> Color:
+	assert(in_atomic_number >= 1 and in_atomic_number <= PeriodicTable.MAX_ATOMIC_NUMBER,
+	"Invalid atomic number %d" % in_atomic_number)
+	if _data[in_atomic_number].set:
+		return _data[in_atomic_number][in_property]
+	# Color is unset, return fallback
+	return _data[0][in_property]
+
+
+func _get_property_list() -> Array[Dictionary]:
+	var properties: Array[Dictionary] = []
+	properties.append({
+		"name": "page",
+		"type": TYPE_INT,
+		"hint": PROPERTY_HINT_RANGE,
+		"hint_string": "1,12,1"
+	})
+	_page = clamp(_page, 1, 12)
+	for i in range((_page-1)*10, (_page*10), 1):
```

Comment:
it seems like we really could use some constants like "NMB_OF_ITEMS_ON_PAGE / MAX_NMB_OF_PAGES / etc" and others to both better self document those numbers and what they are representing as well as to ease the maintenance/changes for people who are not familiar with how this logic works